CMakeLists.txt

```diff
@@ -30,7 +30,8 @@ check_c_compiler_flag("" toolchain_is_ok)
 assert(toolchain_is_ok "The toolchain is unable to build a dummy C file. See CMakeError.log.")
 
 set(CMAKE_EXECUTABLE_SUFFIX .elf)
-set(ZEPHYR_PREBUILT_EXECUTABLE zephyr_prebuilt)
+set(ZEPHYR_PREBUILT_EXECUTABLE zephyr_prebuilt) # Target of 1st linker pass
+set(ZEPHYR_PASS2_EXECUTABLE    zephyr_pass2)    # Target of 2nd linker pass (if applicable)
```

Comment:
You don't know that this is the first or 2nd pass.

You only know that it is the last, and next-to-last pass.

More passes are done when USERSPACE is enabled.

Reply:
Good point. 
How about `ZEPHYR_GEN_EXECUTABLE` or `ZEPHYR_COMPLEX_EXECUTABLE`?

Reply:
ZEPHYR_FINAL_EXECUTABLE and zephyr_final

perhaps. It would re-use the same naming as

linker_pass_final

and communicate correctly that there are no more passes after this (as the build scripts are written today at least).

Reply:
This is still confusing (and just confused me) because for (at least) native_posix the last one is `zephyr_prebuilt`. `zephyr_final` isn't "final" because it's not there at all.
```
if(NOT generated_kernel_files)
  # Use the prebuilt elf as the final elf since we don't have a
  # generation stage.
  set(logical_target_for_zephyr_elf ${ZEPHYR_PREBUILT_EXECUTABLE})
else
```
A good name would relate to *what* happens, not *when* it happens.